File: Ejercicios Veinti21/Ejercicio1.py
import random

teams= ["Manchester United", "Liverpool", "Arsenal", "Chelsea", "Manchester City", "Tottenham"]

table= {"Manchester United" : 0, "Liverpool" : 0, "Arsenal" : 0, "Chelsea" : 0, "Manchester City" : 0, "Tottenham": 0 }
print("-" * 100)
print("MATCH RESULTS")
print("-" * 100)
for x in range(3):
    for i in range(len(teams)):
        team1 = teams[i]
        for j in range(i+1, len(teams)):
            team2 = teams[j]
            gol1 = random.randint(0, 5)
            gol2 = random.randint(0, 5)
            if gol1 > gol2:
                table[team1] += 3
            elif gol1 < gol2:
                table[team2] += 3
            else:
                table[team1] += 1
                table[team2] += 1
            print(f"{team1} {gol1} vs {team2} {gol2} ")

print("-" * 100)
print("FINAL TABLE!!!!")
print("-" * 100)
table_final = sorted(table.items(), key=lambda x: x[1], reverse=True)
for team, result in table_final:
    print(f"{team}: {result}")
# print(table)

            

File: Ejercicio1.swift
/*
1. The Big Six of the English Premier League is composed of six teams: Manchester United, Liverpool, Arsenal, Chelsea, Manchester City, and Tottenham Hotspur.//DONE
 Develop a system that generates between 0, 1, and 3 points randomly for each team. //DONE
 The winner of the Big Six will be the team with the highest accumulated points at the end. 
 Each team will play 3 matches against every opponent. //DONE
 After the matches, the system will display on-screen the team standings from highest to lowest points.
 */

 var Jugadores: [String] = ["Manchester United", "Liverpool", "Arsenal", "Chelsea", "Manchester City", "Tottenham Hotspur"]
var Equipos: [(String , Int)] = []
for jugador in Jugadores{
    let z = Int.random(in: 1...3)
    Equipos.append((jugador, z))
}

for equipo in Equipos{
    print(equipo)
}


func equipo1v2() {

    for a in 1...3{
    for i in 0..<Jugadores.count - 1 {
        for j in (i + 1)..<Jugadores.count {
            let random1 = Int.random(in:0...5)
            let random2 = Int.random(in: 0...5)
  
            print("\(Jugadores[i]) \(random1) vs \(Jugadores[j]) \(random2)")
            if (random1 > random2){
                Equipos[i].1 += 3
            }
            else if(random1 < random2){
                Equipos[j].1 += 3
            }
            else if (random1 == random2){
                Equipos[i].1 += 1
                Equipos[j].1 += 1
            }


        }
    }
    }
}


equipo1v2()


for equipo in Equipos{
    print(equipo)
}

func mayorpuntuacion(){
    print("Tabla---------------")
        let equiposOrdenados = Equipos.sorted { $0.1 > $1.1 }
        print(Equipos)

}



mayorpuntuacion()
//func partido(){
if let ganador = Equipos.first {
        print("El equipo con mayor puntos y por tanto ganador es: \(ganador.0)")
    } else {
        print("A.")
    }
//}

